src/client.ts

```diff
@@ -79,7 +92,7 @@ export class ValaLanguageClient {
     }
 
     get languageServerPath(): string | null {
-        return this.config.languageServerPath
+        return substituteVSCodeVariableInString(this.config.languageServerPath)
```

Comment:
To check: whether `this.config.languageServerPath` defaults to empty (`""`) or `null`.

Reply:
In the debugger, I see that `this.config.languageServerPath` is an empty string when the value is not set in `settings.json`.  The `substituteVSCodeVariableInString` call will, therefore also return an empty string.

I also see that the fallthrough to the  `which.sync` calls works as expected.